### PR TITLE
tweak to the fragment POST logic.

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

File: src/publisher/api_v2_views.py [hunk 1]
```diff
@@ -98,8 +98,7 @@ def article_version(request, id, version):
     "returns the article-json for a specific version of the given article ID"
     authenticated = is_authenticated(request)
     try:
-        # TODO: this seems wrong, should be 'not authenticated'
-        # since you are already at it, test at the HTTP level also the other requests
+        # TODO: test at the HTTP level also the other requests
```
Comment (giorgiosironi) [inline src/publisher/api_v2_views.py]: comment was outdated, right

File: src/publisher/api_v2_views.py [hunk 2]
```diff
@@ -129,12 +128,16 @@ def article_fragment(request, art_id, fragment_id):
         reserved_keys = [XML2JSON]
         ensure(fragment_id not in reserved_keys, "that key is taken")
         with transaction.atomic():
+            article = models.Article.objects.get(manuscript_id=art_id)
+            data = request.data
+            frag, created, updated = fragment_logic.add(article, fragment_id, data, update=True)
```
Comment (giorgiosironi) [inline src/publisher/api_v2_views.py]: right so we only need to do this once as all `av` were pointing to the same article

**Comment by giorgiosironi** [comment]
Seems a good simplification. Does it have an impact on behavior that we can pin down with a new test (initially failing but green after the change)?

**Comment by lsh-0** [comment]
I'm not sure what the original problem was that you were seeing that caused you to touch that code, so I can't test for it.

**Comment by giorgiosironi** [comment]
If anything emerges from end2end I'll port back a smaller test case, no problem.

**Pull request merged**